File: app/assets/javascripts/models/post.js
EmberBlog.Post = DS.Model.extend({
    category: DS.belongsTo('EmberBlog.Category'),
    user: DS.belongsTo('EmberBlog.User'),
    comments: DS.hasMany('EmberBlog.Comment'),

    body: DS.attr('string'),
    categoryId: DS.attr('number'),
    canCreate: DS.attr('boolean'),
    canDestroy: DS.attr('boolean'),
    canUpdate: DS.attr('boolean'),
    tagList: DS.attr('raw'),
    teaser: DS.attr('string'),
    title: DS.attr('string'),
    userId: DS.attr('number'),

    createdAt: DS.attr('date'),
    updatedAt: DS.attr('date'),

    htmlId: function() {
        return "post-" + this.get('clientId');
    }.property('clientId'),

    formattedDate: function() {
        var d = this.get('createdAt').toDateString();
        return moment(d).format('MMMM D, YYYY');
    }.property('createdAt')
});

DS.RESTAdapter.registerTransform('raw', {
    deserialize: function(serialized) {
        return serialized;
    },
    serialize: function(deserialized) {
        return deserialized;
    }
});

File: app/models/post.rb
class Post < ActiveRecord::Base
  acts_as_ordered_taggable
  attr_accessible :body, :category_id, :teaser, :title#, :tag_list
  attr_protected nil

  belongs_to :category
  belongs_to :user
  has_many :comments
  has_many :likes, as: :likable

  validates_presence_of :body
  validates_length_of :teaser, {maximum: 500}
  validates_presence_of :teaser
  validates_presence_of :title

  default_scope order: 'posts.created_at DESC'
end
